tests/foreman/destructive/test_capsule_loadbalancer.py

```diff
@@ -0,0 +1,187 @@
+    haproxy.register_contenthost(module_org.label, haproxy_ak.name)
+    result = haproxy.execute('yum install haproxy policycoreutils-python -y')
+    assert result.status == 0
+    haproxy.execute('rm -f /etc/haproxy/haproxy.cfg')
+    haproxy.session.sftp_write(
+        source=DataFile.DATA_DIR.joinpath('haproxy.cfg'),
+        destination='/etc/haproxy/haproxy.cfg',
+    )
+    haproxy.execute(
+        f'sed -i -e s/CAPSULE_1/{setup_capsules["capsule_1"].hostname}/g '
+        f' --e s/CAPSULE_2/{setup_capsules["capsule_2"].hostname}/g '
+        f' /etc/haproxy/haproxy.cfg'
+    )
+    haproxy.execute('systemctl restart haproxy.service')
+    haproxy.execute('mkdir /var/lib/haproxy/dev')
+    haproxy.session.sftp_write(
+        source=DataFile.DATA_DIR.joinpath('99-haproxy.conf'),
+        destination='/etc/rsyslog.d/99-haproxy.conf',
+    )
+    haproxy.execute('setenforce permissive')
+    result = haproxy.execute('systemctl restart haproxy.service rsyslog.service')
+    assert result.status == 0
+
+    return {'haproxy': haproxy}
+
+
+@pytest.fixture(scope='module')
+def loadbalancer_setup(
+    module_org, content_for_client, setup_capsules, module_target_sat, setup_haproxy
+):
+    """Just to keep the tests arguments clean. Yes it's redundant"""
+    return {
+        'module_org': module_org,
+        'content_for_client': content_for_client,
+        'setup_capsules': setup_capsules,
+        'module_target_sat': module_target_sat,
+        'setup_haproxy': setup_haproxy,
+    }
+
+
+@pytest.mark.tier1
+def test_loadbalancer_register_client_using_ak_to_ha_proxy(loadbalancer_setup, rhel7_contenthost):
+    """Register the client using ak to the capsule
+
+    :id: bd3c2e50-18e2-4be7-8a7f-c32472e17c61
+
+    :Steps:
+        1. run `subscription-manager register --org=Your_Organization \
+            --activationkey=Your_Activation_Key \
+            --serverurl=https://loadbalancer.example.com:8443/rhsm \
+            --baseurl=https://loadbalancer.example.com/pulp/repos`
+
+    :expectedresults: The client should be registered to one of the capsules
+
+    :CaseLevel: Integration
+    """
+    url = f'https://{loadbalancer_setup["setup_haproxy"]["haproxy"].hostname}'
+    server_url = f'{url}:8443/rhsm'
+    base_url = f'{url}/pulp/content/'
+    result = rhel7_contenthost.download_install_rpm(
+        repo_url=f'{url}/pub', package_name='katello-ca-consumer-latest.noarch'
+    )
+    assert result.status == 0
+    rhel7_contenthost.register_contenthost(
+        org=loadbalancer_setup['module_org'].label,
+        activation_key=loadbalancer_setup['content_for_client']['client_ak'].name,
+        serverurl=server_url,
+        baseurl=base_url,
+    )
+    result = rhel7_contenthost.execute('rpm -qa |grep katello-ca-consumer')
+    assert (
+        loadbalancer_setup['setup_capsules']['capsule_1'].hostname
+        or loadbalancer_setup['setup_capsules']['capsule_2'].hostname in result.stdout
+    )
```

Comment:
You should also check if the content host appears in satellite, and additionally check if its through one of the capsules.

Reply:
I have added the check where the host lists on the satellite. :)

As for which capsule, I think the `<capsule>-katello-ca-consumer-latest.noarch` on the contenthost is the only way to check which capsule the host is being registered to. 
When we do `hammer host info --id <ID>`, we can see `Subscription Information`, and that's the only info where you could tell where the host is registered through, but since we do the following on the content host
```
subscription-manager register --org=Your_Organization \
            --activationkey=Your_Activation_Key \
            --serverurl=https://loadbalancer.example.com:8443/rhsm \
            --baseurl=https://loadbalancer.example.com/pulp/content`
```
this changes the rhsm.conf, and the contents of this file gets uploaded in the `Subscription Information` according to serverurl and baseurl, which will be the loadbalancer. I think it's a very valid RFE on this scenario.

Reply:
Sure! Thanks for in detail reply. and yes, we could surely raise an RFE in BZ for it.